install/migration/migration.py

```diff
@@ -71,9 +73,18 @@ def application(environ, start_response):
     if environ.get('REQUEST_METHOD', None) != 'POST':
         return wsgi_redirect(start_response, 'index.html')
 
+    content_type = environ.get('CONTENT_TYPE', '').lower()
+    if not content_type.startswith('application/x-www-form-urlencoded'):
+        return bad_request(start_response)
+
     form_data = cgi.FieldStorage(fp=environ['wsgi.input'], environ=environ)
     if 'username' not in form_data or 'password' not in form_data:
-        return wsgi_redirect(start_response, 'invalid.html')
+        return bad_request(start_response)
+
+    status = '200 Success'
+    response_headers = []
+    result = 'error'
+    response = b''
```

Comment:
Not sure why we should have status '200 Success' and result 'error' here.

Reply:
'200 Success' is here because we've already fulfilled the client's info from request ( HTTP method, content type, params (username and password) ) . After a successful parsing and validation of that we cannot reply with 4xx. 
```python
    if environ.get('REQUEST_METHOD', None) != 'POST':                           
        return wsgi_redirect(start_response, 'index.html')                      
                                                                                
    content_type = environ.get('CONTENT_TYPE', '').lower()                      
    if not content_type.startswith('application/x-www-form-urlencoded'):        
        return bad_request(start_response)                                      
                                                                                
    form_data = cgi.FieldStorage(fp=environ['wsgi.input'], environ=environ)        
    if 'username' not in form_data or 'password' not in form_data:              
        return bad_request(start_response)                                      
                                                                                
    status = '200 Success'                                                      
    response_headers = []                                                       
    result = 'error'                                                            
    response = b''
``` 
```result = 'error'``` is just the default value for this variable. The other possible ones: 'ok', 'invalid-password', 'migration-error'.
So the result of HTTP request - 200, but the result of migration is one of the 3 possible.

Reply:
Ok, sounds good, just wanted an explanation for the mix.